pkg/controller/oneagent-utils/verification_test.go

```diff
@@ -62,3 +65,29 @@ func TestVerifySecret(t *testing.T) {
 		assert.NoError(t, err)
 	}
 }
+
+func TestBuildDynatraceClientWithCustomToken(t *testing.T) {
```

Comment:
Suggest to add an additional test for covering the erroneous use case.

Reply:
This is the erroneous use case, but I can add one for the normal one (i.e., token and secret have the same name.)

Reply:
This one asserts `NoError`. I would add one which asserts an `Error` to verify if a faulty configuration gets handled accordingly.

Reply:
Ah, ok, fair, will do.